src/fknni/faiss/faiss.py

```diff
@@ -32,87 +31,156 @@ def __init__(
                       where closer neighbors have a higher influence on the imputation.
             index_factory: Description of the Faiss index type to build. Defaults to 'Flat'.
         """
-        super().__init__()
+        if n_neighbors < 1:
+            raise ValueError("n_neighbors must be at least 1.")
+        if strategy not in {"mean", "median", "weighted"}:
+            raise ValueError("Unknown strategy. Choose one of 'mean', 'median', 'weighted'")
+
         self.missing_values = missing_values
         self.n_neighbors = n_neighbors
         self.metric = metric
         self.strategy = strategy
         self.index_factory = index_factory
+        self.min_data_ratio = 0
+        self.X_full = None
+        self.features_nan = None
+        self.min_data_ratio = 0.25
+        super().__init__()
 
-    def fit(self, X: np.ndarray | pd.DataFrame, *, y: np.ndarray | None = None) -> FaissImputer:
-        """Fits the FaissImputer to the provided data.
+    @overrides
+    def fit_transform(self, X: np.ndarray, y=None, **fit_params) -> ndarray[Any, dtype[Any]] | None:
+        """Imputes missing values in the data using the fitted Faiss index.
 
         Args:
             X: Input data with potential missing values.
             y: Ignored, present for compatibility with sklearn's TransformerMixin.
 
-        Raises:
-          ValueError: If any parameters are set to an invalid value.
+        Returns:
+            Data with imputed values as a NumPy array of the original data type.
```

Comment:
I could see us eventually single dispatching this and attempting to return the same array type as we put in. However, this would greatly expand the scope of everything here so for now this is random useless comment haha.